### PR TITLE
Automobile Sensor

### PR DESCRIPTION
**Pull request by Manish-Fexle-May2022**
Assignment Automobile sensor

### CODE REVIEW

File: src/classes/OpportunityHelper.cls [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+            for(Automobile_Sensor__c recAutomobile : mapAutomobileSensor.values()){
+                if(recAutomobile.Opportunity__c  == opp.Id){
+                    //<td> - Defines a cell in a table or Used for creates standard data cell in HTML table
+                    tableRow = tableRow + '<tr><td>Sensor</td><td>'+recAutomobile.Name+'</td><td>Sensor inserted under installation account : '+recAutomobile.Account__r.Name+'</td></tr>';
+                }        
+            }
+            for(Automobile_Presence_Sensor__c recAutomobile : mapAutomobilePresenceSensor.values()){
+                if(recAutomobile.Opportunity__c  == opp.Id){
+                    tableRow = tableRow + '<tr><td>Presence Sensor</td><td>'+recAutomobile.Name+'</td><td>Presence Sensor inserted under installation account : '+recAutomobile.Account__r.Name+'</td></tr>';
```
Comment (shubhamsharmafexletraining) [inline src/classes/OpportunityHelper.cls:201]: Break lines

**Review by shubhamsharmafexletraining — CHANGES_REQUESTED** [review]

File: src/classes/OpportunityHelper.cls [hunk 2]
```diff
@@ -0,0 +1,261 @@
+public class OpportunityHelper {
+    public static Automobile_Sensor__c recordSensor(Boolean isAfter,List<String> NameList){
```
[the remark this answers is not in the document]
Comment (Manish-Fexle-May2022) [reply src/classes/OpportunityHelper.cls]: Sir I am Using Boolean isAfter for Not Creating Duplicate Records when Updating Opportunity stage

[the remark this answers is not in the document]
Comment (shubhamsharmafexletraining) [reply src/classes/OpportunityHelper.cls]: This is not the correct logic. You are passing trigger.isUpdate to the helper class method and on update you are not creating/updating any sensor. This is the wrong approach. sensors records should be create/update/delete when opportunity create/Update and having the appropriate stage. this whole logic is wrong.

**Comment by shubhamsharmafexletraining** [comment]
Closing the pull request for wrong logic. please raise another pull request with correct logic.

**Pull request closed without merging**